scripts/tokenRefresh.js

```diff
@@ -52,13 +50,9 @@ async function tokenRefresh() {
 			await delay(200);
 			responseObject = await response.json();
 			if (Object.hasOwn(responseObject, "statusDescription")) {
-				if (!responseObject.statusDescription.includes("The Token field is required.")) 
-					return tokenRefresh();	
-			}
-			else
-				return tokenRefresh();
-		} else 
-			openLoginMenu();
+				if (!responseObject.statusDescription.includes("The Token field is required.")) return tokenRefresh();
```

Comment:
para que é que vês se deu erro por falta de token em vez de veres logo à partida se estás a mandar um token?

Reply:
estou a perceber, mas ao mesmo tempo não estou a perceber exatamente

o objetivo aqui é não dar outro token refresh apenas se existir um statusDescription e esse statusDescription não for o "The Token field is required."

haverá forma de condensar isto de maneira mais elegante?

Reply:
ao mesmo tempo tens um else para se for esse o caso ele dar refresh ao token. O que eu tou a dizer é que se pode evitar esse pedido e ver se o token existe antes de o mandar, dando refresh antes deste pedido. Dessa forma sabemos que o erro que venha depois não é por falta de token